docs/rbac.rst

```diff
@@ -0,0 +1,266 @@
+         "rpm.view_ulnremote",
+      ],
+      "locked": true
+    }
+
+.. important::
+
+   Please note there are two types of Remotes with their individual permissions.
+
+
+RPM Viewer Role (rpm.viewer)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A universal role allows you to view all content within the RPM plugin.
+
+.. code-block:: shell
+   :caption: Universal RPM viewer role
+
+    {
+      "pulp_href": "/pulp/api/v3/roles/d1e7bae0-2363-45e2-815c-d721ecc3133c/",
+      "pulp_created": "2022-04-07T12:41:11.144883Z",
+      "name": "rpm.viewer",
+      "description": null,
+      "permissions": [
+        "rpm.view_rpmalternatecontentsource",
+        "rpm.view_rpmdistribution",
+        "rpm.view_rpmpublication",
+        "rpm.view_rpmremote",
+        "rpm.view_rpmrepository",
+        "rpm.view_ulnremote",
+      ],
+      "locked": true
+    }
+
+
+Creating Roles
+--------------
+
+.. note::
+
+    Roles shipped by the RPM plugin mentioned above are locked and cannot be modified or removed.
+
+To create a new role, you need to specify its name and permissions to use.
+To find out which permissions you need for a new role, you can list an endpoints policy with the following command:
+
+.. code-block:: shell
+   :caption: List an access policy of the repository endpoint
+
+    pulp access-policy show --viewset-name "repositories/rpm/rpm"
+
+**Example 1: RPM Repository Creator**
+
+.. code-block:: shell
+   :caption: Example of the RPM repository creator role
+
+    pulp role create --name "rpm_repo_creator" \
+        --permission "rpm.add_rpmrepository" \
+        --permission "rpm.view_rpmrepository" \
+        --permission "rpm.sync_rpmrepository" \
+        --permission "rpm.modify_rpmrepository" \
+        --permission "rpm.change_rpmrepository" \
+        --permission "rpm.delete_rpmrepository" \
+        --permission "rpm.add_rpmremote" \
+        --permission "rpm.change_rpmremote" \
+        --permission "rpm.delete_rpmremote" \
+        --permission "rpm.view_rpmremote"
+
+**Example 2: RPM Publisher**
+
+.. code-block:: shell
+   :caption: Example of the RPM publisher role
+
+    pulp role create --name "rpm_publisher" \
+        --permission "rpm.view_rpmrepository" \
+        --permission "rpm.add_rpmpublication" \
+        --permission "rpm.add_rpmdistribution"
+
+.. code-block:: shell
+   :caption: Globally assign these roles to user
+
+    pulp user role-assignment add --username "bob" --role "rpm_repo_creator" --object ""
+    pulp user role-assignment add --username "jack" --role "rpm_publisher" --object ""
+
+.. note::
+
+   You can only assign roles at the object level if the role contains at least one permission applicable for that object.
+
+Edit Access Policy
+------------------
+
+It is possible to edit the `access policy` for an endpoint.
+
+.. code-block:: shell
+   :caption: Change the access policy of all actions not requiring any permission.
+
+    pulp access-policy update \
+        --viewset-name "remotes/rpm/rpm" \
```

Comment:
``viewset-name`` is already in the CLI and already used in a few more plugins so changing it wouldn't be so easy.

Reply:
That is unfortunate.  @mdellweg Do you think this would be a worthwhile change, or do you think the cost would outweigh the benefit?  If we have to carry around the old commands indefinitely it's probably not worth it.

Reply:
The thing that i was told cannot change is the parameter in the REST api. The cli is just following the bad naming for consistency and less confusion and i'd rather keep it that way.

Reply:
+1